projects/gateway2/deployer/values.go

```diff
@@ -163,3 +166,15 @@ type helmAws struct {
 	StsClusterName                  *string `json:"stsClusterName,omitempty"`
 	StsUri                          *string `json:"stsUri,omitempty"`
 }
+
+type helmTlsSecretData struct {
+	CaCert  []byte `json:"caCert,omitempty"`
+	TlsCert []byte `json:"tlsCert,omitempty"`
+	TlsKey  []byte `json:"tlsKey,omitempty"`
+}
+
+type helmMtlsConfig struct {
+	Enabled      *bool              `json:"enabled,omitempty" desc:"Enables internal mtls authentication"`
+	RenderSecret *bool              `json:"renderSecret,omitempty" desc:"If true, the deployer will render the mtls secrets. Used for creating the intial sceret resource for mointoring before the full config is available"`
```

Comment:
looks like this is not used (and redundant w/Enabled)

Reply:
This is used here: https://github.com/solo-io/gloo/blob/sheidkamp/k8s-gw-mtls-deployer/projects/gateway2/deployer/deployer.go#L137-L142 during setup.

This ensures that the Secret resource is rendered and watched. If we use the `Enabled` flag for this, it does all the other rendering, for example the sds container, and we don't have the data to support it resulting in null pointers.

We could update the Helm code to handle null sds data, but in every other case than `GetGvksToWatch` missing data would be indicative of an error/problem, so I like having a separate flag to make it clear what we are trying to do.